src/libraries/System.Linq.Expressions/src/System/Dynamic/Utils/DelegateHelpers.cs

```diff
@@ -23,9 +23,21 @@ private static class DynamicDelegateLightup
                 = CreateObjectArrayDelegateInternal();
 
             private static Func<Type, Func<object?[], object?>, Delegate> CreateObjectArrayDelegateInternal()
-                => Type.GetType("Internal.Runtime.Augments.DynamicDelegateAugments")!
-                    .GetMethod("CreateObjectArrayDelegate")!
-                    .CreateDelegate<Func<Type, Func<object?[], object?>, Delegate>>();
+            {
+                // This is only supported by NativeAOT which always expects CanEmitObjectArrayDelegate to be false.
```

Comment:
Instead of all this, it would be better to use `UnsafeAccessor` to access the internal CoreLib helper, similar to how it was done for other internal CoreLib helper here: https://github.com/dotnet/runtime/pull/89706/

Reply:
`ForceAllowDynamicCodeLightup` below can be switched to `UnsafeAccessor` too.

I think this cleanup can wait for .NET 9. Could you please open a tracking issue for it?

Reply:
I think this will need https://github.com/dotnet/runtime/issues/90081 first.

Reply:
It does not need #90081. We can access the internal method on the Delegate type and delete `Internal.Runtime.Augments.DynamicDelegateAugments` that just forwards to the internal method on the Delegate type.

(`Internal.Runtime.Augments.DynamicDelegateAugments` forwarder exists just to avoid having the internal method exposed as public.)